prime-router/src/main/kotlin/serializers/CsvSerializer.kt

```diff
@@ -147,36 +147,27 @@ class CsvSerializer(val metadata: Metadata) : Logging {
     }
 
     /**
-     * Reads an internal format document (one that is in CSV, but matching the internal "kitchen sink" schema)
-     * and returns the Report object for the file.
-     * @param useDefaultsForMissing if a field is missing that the Report object expects, this instructs the
-     * function to use the default value provided by the schema (or empty string) if it doesn't exist. This was
-     * added because some older files might need to be reprocessed, and this allows that to happen without error
-     * due to a missing column
+     * Reads an internal format document from [input] using the schema [schemaName]and returns the Report object for
+     * the report for a given [destination]
+     * @param sources the list of sources for this report
+     * @param blobReportId the report ID
+     * @return the report
      */
     fun readInternal(
         schemaName: String,
         input: InputStream,
         sources: List<Source>,
         destination: Receiver? = null,
-        blobReportId: ReportId? = null,
-        useDefaultsForMissing: Boolean = false
+        blobReportId: ReportId? = null
     ): Report {
-        // find our schema
         val schema = metadata.findSchema(schemaName) ?: error("Internal Error: invalid schema name '$schemaName'")
-        // get our rows
-        val rows = if (useDefaultsForMissing) {
-            // walk through the rows in the file with the header included
-            csvReader().readAllWithHeader(input).map {
-                // for each element name, if it doesn't exist in the map, then we add it with a default
-                // doing it this ways means that even if someone adds a new element in the middle of the schema
-                // (please don't), this should still be okay and it won't necessarily break
-                schema.elements.map { element ->
-                    it.getOrDefault(element.name, element.default ?: "")
-                }
+        val rows = csvReader().readAllWithHeader(input).map {
```

Comment:
Is this going to cause downstream effects / are there specific fields that are simply invalid if they did not have a user-provided value?

Reply:
No.  All empty fields in a report are always an empty string.

Reply:
Will this work if the new field has cardinality: ONE but no default?

Reply:
No, because the initial serializer will catch that an error out the report.  It is the reason cardinality or mappers are not checked here.